### PR TITLE
QOL-8904 - Added global modal popup feature for wysiwyg editor on matrix

### PR DESCRIPTION
**Pull request by manjeetkgov**

### CODE REVIEW

File: js/on-demand-scripts/qg-modal.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  
+        if (button.hasClass('image-modal')) {
+          modal.addClass('modal-img');
+          modal.find('.modal-content').html('<div class="modal-body">').prepend('<div class="modal-header">' + close + '</div>');
+          modal.find('.modal-body').html('<img src="' + url + '">');
+          return;
+        } // Load via ajax (from another page).
+  
+  
+        $.ajax({
```
Comment (devonpis) [inline js/on-demand-scripts/qg-modal.js:94]: better check `button.data('href')` is not undefined before fire the ajax call

File: js/on-demand-scripts/qg-modal.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+          url: button.data('href'),
+          // Load the content in to the modal.
+          success: function success(data) {
```
Comment (devonpis) [inline js/on-demand-scripts/qg-modal.js:97]: is there any CORS error if data is loaded from different domain?

Comment (manjeetkgov) [reply js/on-demand-scripts/qg-modal.js:97]: This feature will be available on the same domain (qld.gov.au)

File: js/on-demand-scripts/qg-modal.js (hunk 1 above)
Comment (manjeetkgov) [reply js/on-demand-scripts/qg-modal.js:94]: All links will have a href to trigger the modal. If there is no href associated modal code will not generate.

**Review by devonpis — APPROVED** [review]

**Pull request merged**